tos/middleware.py

```diff
@@ -19,8 +19,8 @@ class UserAgreementMiddleware(MiddlewareMixin):
     Some middleware to check if users have agreed to the latest TOS
     """
 
-    def __init__(self, get_response=None):
-        self.get_response = get_response
+    def __init__(self, get_response):
+        super().__init__(get_response)
```

Comment:
As this is written, couldn't it simply be omitted?

But i'm also wondering if this is no longer backwards compatible. If someone bumps to this with no changes in their codebase, they'll get hit with a missing argument error.

Can we keep the override and check django version here, and if pre 4.0 allow None to be passed? I want to make sure nobody is affected by this when they upgrade.

An additional test, without passing an arg to the UserAgreementMiddleware would help cover this.

Reply:
The middleware currently adheres to the outdated style (pre-Django version 1.10), and it may be advantageous to modernize it. As it is a middleware, it is unlikely that anyone calls it directly, but rather it is utilized in the chain of registered middlewares. Consequently, adapting or omitting the constructor should not pose any complications, or am I missing something?